### ISSUE TITLE
Button to exclude repository from repository status page

### ISSUE THREAD

**Issue by mike-r-mclaughlin** [sourcegraph/sourcegraph#40504]
**Requested by:** https://github.com/sourcegraph/accounts/issues/3

#### Feature request description

Add a button/link to the repository status page to allow the user to immediately add a repo (or repos) to the exclude list. 

#### Is your feature request related to a problem? If so, please describe.

When bulk adding repositories, you will often find many repositories that fail for a variety of reasons.  And, the solution (once you determine that it's failing for a reason you can't resolve - things like corrupt repo, etc.) is to go over to the code host configuration and add it to the exclude list.  This feature would make it easier to do that from the repository status page.

**Comment by mrnugget** [comment]
Notes from grooming meeting:

- Only for site-admins
- Add "code hosts" section to repositories settings page
- [ ] Add GraphQL mutation to `excludeRepoFromExternalService(externalService: ID!, repo: ID!)`
  - update config
  - trigger sync
- [ ] Add button to repo settings page (https://sourcegraph.sourcegraph.com/github.com/microsoft/calculator/-/settings) to exclude repo
   - [ ] Next to each code host show button 
   - [ ] Do not show button if repo is not excludable (`isExcludable` to GraphQL type `Repo`?)
   - [ ] disable buttons if EXTSVC_CONFIG_FILE is set
- [ ] Redirect to external service config page after clicking button

**Comment by mike-r-mclaughlin** [comment]
> Is a repo is sourced from multiple code hosts, should this button exclude it from all the code host configs?

Yes.  For both https://github.com/sourcegraph/accounts/issues/3, https://github.com/sourcegraph/accounts/issues/6716 the reason for this request is because there is something wrong with the repository (spotted on the status page) and they want to exclude it because of that (usually because it's empty repo or because it's too large and causing errors like [#1245](https://github.com/sourcegraph/customer/issues/1245)).

**Comment by sashaostrikov** [comment]
Further ideas, related to the issue:
- If there are many code hosts per repo, then after clicking the "exclude" button, we can show a message that code host config is updated and provide a link to it instead of making a redirect
- Add a spinner to [code hosts page](https://sourcegraph.sourcegraph.com/site-admin/external-services?visible=19) in case of in progress sync of a code host
  - This requires a GraphQL API update
